src/javascript/reducers/agreementList.js

```diff
@@ -0,0 +1,141 @@
+// @flow
+import type {
+  RequestPendingAction,
+  BuyerSentAction,
+  SellerSentAction,
+  SendFailedAction,
+  ConfirmedAction,
+  ConfirmedFailedAction,
+  TransactionCanceledAction,
+  TransactionCancelFailedAction
+} from '../actions/agreementList';
+
+import type {
+  AgreementListType,
+  AgreementType
+} from '../types';
+
+type Action =
+  | RequestPendingAction
+  | BuyerSentAction
+  | SellerSentAction
+  | SendFailedAction
+  | ConfirmedAction
+  | ConfirmedFailedAction
+  | TransactionCanceledAction
+  | TransactionCancelFailedAction;
+
+const INITIAL_STATE: AgreementListType = {
+  agreements: {
+    "asdfasdfas": {
+      creationDate: 1493690737487,
+      position: "buyer",
+      price: 1234,
+      buyer: "123.12.23.21",
+      seller: "1231.123.12.1231",
+      buyerPaid: true,
+      buyerPaidDate: 1493690737488,
+      sellerPaid: true,
+      sellerPaidDate: 1493690737497,
+      transactionComplete: false,
+      transactionCompleteDate: 0,
+      error: "",
+      canceled: false,
+      canceledDate: 1493690737506,
+      requestPending: false,
+    },
+  },
+  allTransactions: ["asdfasdfas"],
+};
+export default function agreementListReducer(state: AgreementListType = INITIAL_STATE, action: Action): AgreementListType {
+  const transaction: AgreementType = state[action.payload];
+  switch (action.type) {
+    // TODO: figure out if putting "error:""" is
+    // actually the way to do it
+    case 'REQUEST_PENDING':
+      return {
+        ...state,
+        transaction: {
```

Comment:
Also what do you think about renaming this variable from `const transaction: AgreementType` to `const agreement: AgreementType`?

Especially since "transaction" has special meaning in Ethereum